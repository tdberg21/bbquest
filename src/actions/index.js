export const addRestaurants = (restaurants) => ({
  type: 'ADD_RESTAURANTS',
  restaurants
});

export const loginUser = (username, id) => ({
  type: 'LOGIN_USER',
  username,
  id
});

