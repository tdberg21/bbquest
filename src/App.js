import React, { Component } from 'react';
import './App.css';
import { fetchRestaurantData } from './helpers/apiCalls';
import { apiKey } from './helpers/apiKey';
import { mockResponse } from './helpers/mockdata';
import { scrubRestaurants } from './helpers/dataCleaners';
import { addRestaurants } from './actions';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';
import LoginForm from './containers/LoginForm/LoginForm';
import Header from './containers/Header';

class App extends Component {

  componentDidMount = async () => {
    // const restaurants = await fetchRestaurantData(apiKey);
    const cleanRestaurants = scrubRestaurants(mockResponse);
    this.props.addRestaurants(cleanRestaurants);
    console.log(cleanRestaurants);
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Route exact path='/login' component={LoginForm} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  restaurants: state.restaurants
});

export const mapDispatchToProps = (dispatch) => ({
  addRestaurants: (restaurants) => dispatch(addRestaurants(restaurants))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
