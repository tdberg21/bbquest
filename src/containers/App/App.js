import React, { Component } from 'react';
import './App.css';
// import { fetchRestaurantData } from './helpers/apiCalls';
// import { apiKey } from './helpers/apiKey';
import { mockResponse } from '../../helpers/mockdata';
import { scrubRestaurants } from '../../helpers/dataCleaners';
import { addRestaurants } from '../../actions';
import { connect } from 'react-redux';
import { Route, withRouter } from 'react-router-dom';
import LoginForm from '../LoginForm/LoginForm';
import Header from '../Header/Header';
import CardContainer from '../CardContainer/CardContainer';
import SearchForm from '../SearchForm/SearchForm';
import SignUpForm from '../SignUpForm/SignUpForm';
import RestaurantDetails from '../RestaurantDetails/RestaurantDetails';
import VisitedForm from '../VisitedForm/VisitedForm';

export class App extends Component {

  // componentDidMount = async () => {
  //   // const restaurants = await fetchRestaurantData(apiKey);
  //   const cleanRestaurants = scrubRestaurants(mockResponse);
  //   this.props.addRestaurants(cleanRestaurants);
  //   console.log(cleanRestaurants);
  // }

  render() {
    return (
      <div className="App">
        <Header />
        <Route path='/login' component={LoginForm} />
        <Route path='/search' component={SearchForm} />
        <Route exact path='/' component={SearchForm} />
        <Route exact path='/restaurants' component={CardContainer} />
        <Route path='/signup' component={SignUpForm} />
        <Route path='/restaurants/:name' render={({ match }) => {
          let restaurant = this.props.restaurants.find(restaurant => restaurant.name === match.params.name);
          return <RestaurantDetails {...restaurant} />;
        }} />
        <Route path='/restaurants/:name/review' component={VisitedForm}/>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  restaurants: state.restaurants
});

export const mapDispatchToProps = (dispatch) => ({
  addRestaurants: (restaurants) => dispatch(addRestaurants(restaurants))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
