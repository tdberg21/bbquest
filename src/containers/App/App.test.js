/*eslint-disable no-undef*/

import React from 'react';
import { App, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { addRestaurants, logOutUser, addVisited } from '../../actions';

describe('APP TESTS', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should push you back to restaurants page if there is a duplicate', () => {
    const mockVisited = [{yelpId:1}];
    const mockPush = jest.fn();
    const mockHistory = {push: mockPush};
    const wrapper = shallow(<App visited={mockVisited} history={mockHistory}/>);
    
    wrapper.instance().checkVisited(1);
    
    expect(mockPush).toHaveBeenCalled();
  });

  it('should set state if there is no duplicate', () => {
    const mockVisited = [{ yelpId: 1 }];
    const mockPush = jest.fn();
    const mockHistory = { push: mockPush };
    const wrapper = shallow(<App visited={mockVisited} history={mockHistory} />);

    wrapper.instance().checkVisited(2);

    expect(wrapper.state('yelpId')).toEqual(2);
  });

  it('should call addVisited when addRestaurantToDB is invoked', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      ok: true,
      json: () => Promise.resolve({
        results: ['PeeWees']
      })
    }));
    const mockUser = {id: 9};
    const mockAddVisited = jest.fn();
    const mockRestaurants = [{ id: 1, name: 'PeeWees' }, { id: 2, name: 'Moes' }];
    const wrapper = shallow(<App restaurants={mockRestaurants} user={mockUser} addVisited={mockAddVisited}/>);
    wrapper.setState({
      yelpId: 1
    });

    await wrapper.instance().addRestaurantToDatabase(2, 'taco', 'june', 'food');

    await expect(mockAddVisited).toHaveBeenCalled();
  });

  it('should find a restaurant', () => {
    const mockRestaurants = [{id: 1, name: 'PeeWees'}, {id: 2, name: 'Moes'}];
    const wrapper = shallow(<App restaurants={mockRestaurants}/>);
    const expected = {id: 1, name: 'PeeWees'};
    const results = wrapper.instance().findRestaurant(1);
    
    expect(results).toEqual(expected);
  });

  it('should call logOutUser when logOut is invoked', () => {
    const mockLogOut = jest.fn();
    const wrapper = shallow(<App logOut={mockLogOut} addVisited={jest.fn()}/>);

    wrapper.instance().logOut();
    expect(mockLogOut).toHaveBeenCalled();
  });

  it('should call addVisited when logOut is invoked', () => {
    const mockAddVisited = jest.fn();
    const wrapper = shallow(<App logOut={jest.fn()} addVisited={mockAddVisited} />);

    wrapper.instance().logOut();
    expect(mockAddVisited).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    
    it('calls dispatch with addRestaurants action', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addRestaurants(['PeeWees']);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addRestaurants(['PeeWees']);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with logOutUser action', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = logOutUser();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.logOut();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with addVisited action', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addVisited('PeeWees');
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addVisited('PeeWees');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
