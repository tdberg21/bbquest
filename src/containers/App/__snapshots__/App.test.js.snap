// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP TESTS should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          logOutUser={[Function]}
          user={undefined}
          visited={undefined}
/>,
        <Route
          component={[Function]}
          path="/login"
/>,
        <Route
          component={[Function]}
          path="/search"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/restaurants"
/>,
        <Route
          component={[Function]}
          path="/signup"
/>,
        <Route
          path="/restaurants/:name"
          render={[Function]}
/>,
        <Route
          path="/restaurants/:name/review"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          path="/visited"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "logOutUser": [Function],
          "user": undefined,
          "visited": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/restaurants",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/restaurants/:name",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/restaurants/:name/review",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/visited",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            logOutUser={[Function]}
            user={undefined}
            visited={undefined}
/>,
          <Route
            component={[Function]}
            path="/login"
/>,
          <Route
            component={[Function]}
            path="/search"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/restaurants"
/>,
          <Route
            component={[Function]}
            path="/signup"
/>,
          <Route
            path="/restaurants/:name"
            render={[Function]}
/>,
          <Route
            path="/restaurants/:name/review"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            path="/visited"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "logOutUser": [Function],
            "user": undefined,
            "visited": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/restaurants",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/restaurants/:name",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/restaurants/:name/review",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/visited",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
